scrape/scrape.go

```diff
@@ -685,12 +709,18 @@ func (c *scrapeCache) iterDone(flushCache bool) {
 			}
 		}
 		c.metaMtx.Lock()
+		var metaSize int
 		for m, e := range c.metadata {
 			// Keep metadata around for 10 scrapes after its metric disappeared.
 			if c.iter-e.lastIter > 10 {
 				delete(c.metadata, m)
+				continue
 			}
+			metaSize += len(m) + e.size()
 		}
+
+		targetMetadataCacheBytes.WithLabelValues(jobName).Set(float64(metaSize))
```

Comment:
I've used `jobName` as the identifier of a target given we do something similar in this file - I tried using the target's labels, but that seems quite tricky because of relabelling. Happy to be guided on what would be best!

Reply:
sorry @brian-brazil, I'm not sure I follow. Wouldn't the target(s) all belong to the same job under the current grouping? 

Taking another pass at the code, I think the `job` does not uniquely identify the target. Per the [group struct](https://github.com/prometheus/prometheus/blob/master/discovery/targetgroup/targetgroup.go#L25-L27), it seems I should be using the address label?

Reply:
A scrape config usually has more than one target.

Reply:
But the scrapeCache is per target, isn't it?

We have to calculate the value differently / somewhere else, if we really want the total size.

Reply:
I would hope for simplicity that we add that together with `up` metric

Reply:
Yep @roidelapluie, your comment makes perfect sense. I'll do that.

Reply:
There's quite a high bar to adding a new sample to every single scrape that happens, that adds storage cost for every single Prometheus usage.

This belongs on /metrics, but as it stands not as a scrape metric.

Reply:
mmh yeah indeed brian. However then it makes it like we will see it only per job?

Reply:
Yes, the same others on /metrics. We'd usually not even break something like this out by job.

Reply:
Indeed I am not sure I want 300 new metrics for this